tests/integrations/raftstore/test_scale_pool.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use engine_traits::CF_DEFAULT;
+use kvproto::raft_cmdpb::RaftCmdResponse;
+use libc::{getpid, pid_t};
+use procinfo::pid;
+use raftstore::Result;
+use std::collections::HashMap;
+use std::time::Duration;
+use test_raftstore::*;
+use tikv_util::{metrics::get_thread_ids, HandyRwLock};
+
+fn put_with_timeout<T: Simulator>(
+    cluster: &mut Cluster<T>,
+    key: &[u8],
+    value: &[u8],
+    timeout: Duration,
+) -> Result<RaftCmdResponse> {
+    let mut region = cluster.get_region(key);
+    let region_id = region.get_id();
+    let req = new_request(
+        region_id,
+        region.take_region_epoch(),
+        vec![new_put_cf_cmd(CF_DEFAULT, key, value)],
+        false,
+    );
+    cluster.call_command_on_node(0, req, timeout)
+}
+
+#[test]
+fn test_increase_pool() {
+    let mut cluster = new_node_cluster(0, 1);
+    cluster.cfg.raft_store.store_batch_system.pool_size = 1;
+    cluster.cfg.raft_store.apply_batch_system.pool_size = 1;
+    cluster.pd_client.disable_default_operator();
+    let fp1 = "poll";
+    fail::cfg(fp1, "3*pause").unwrap();
```

Comment:
Please add some comments for the test logic like the purpose of the fail point and what do we expect after the config updated, etc.